staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -693,8 +696,26 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 				return nil, err
 			}
 		}
-		// TODO: Once we get rid of /healthz consider changing this to post-start-hook.
-		err := s.addReadyzChecks(healthz.NewInformerSyncHealthz(c.SharedInformerFactory))
+		err = s.AddPostStartHook("generic-apiserver-informers-synced", func(context PostStartHookContext) error {
+			return wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
+				stopCh := make(chan struct{})
+				close(stopCh) // close stopCh to force checking if informers are synced now.
+				informersByStarted := map[bool][]string{}
+
+				for informerType, started := range c.SharedInformerFactory.WaitForCacheSync(stopCh) {
```

Comment:
this WaitForCacheSync is reentrant? We might have more go routines calling this.

Reply:
yes, it is. The function creates and initializes a map by calling `HasSynced` on each registered informer. `HasSynced` is safe for concurrent access because it acquires a lock.